src/cmds/qstat.c

```diff
@@ -302,47 +361,163 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
+			if (keyvalpair == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
 			}
-			while (*c) {
-				putchar(*c++);
-				if (++start > 78) {
-					start = 8;
-					printf("\n\t");
+			while (*value) {
+				/* value is split based on comma and each key-value is stored in keyvalpair.
+				 * If the value contains an escaped comma, only the comma is copied to keyvalpair.  */
+				while (!(*value == *comma || *value == '\0')) {
+					if (*value == ESC_CHAR && *(value + 1) == *comma)
+						value++;
+					keyvalpair[i++] = *value++;
+				}
+				keyvalpair[i] = '\0';
+				key = strtok_r(keyvalpair, "=", &val);
+				if (add_json_node(JSON_VALUE, JSON_NULL, key, val) == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				if (*value != '\0') {
+					i = 0;
+					value++;
+				}
+			}
+			if (add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL) == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
+			}
+			free(keyvalpair);
+		} else {
+			if (resource) {
+				if (prev_resc_name && strcmp(prev_resc_name, name) != 0) {
+					if (add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL)
+							== NULL) {
+						fprintf(stderr, "qstat: out of memory\n");
+						exit(1);
+					}
+					prev_resc_name = NULL;
+				}
+				if (prev_resc_name == NULL || strcmp(prev_resc_name, name) != 0) {
+					if (add_json_node(JSON_OBJECT, JSON_NULL, name, NULL)
+							== NULL) {
+						fprintf(stderr, "qstat: out of memory\n");
+						exit(1);
+					}
+					prev_resc_name = name;
+				}
+				if (add_json_node(JSON_VALUE, JSON_NULL, resource, value)
+						== NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+			} else {
+				if (prev_resc_name) {
+					if (add_json_node(JSON_OBJECT_END, JSON_NULL, NULL, NULL)
+							== NULL) {
+						fprintf(stderr, "qstat: out of memory\n");
+						exit(1);
+					}
+					prev_resc_name = NULL;
+				}
+				if (add_json_node(JSON_VALUE, JSON_NULL, name, value) == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
 				}
 			}
 		}
-		if ((c = strtok((char *)0, comma)) != NULL) {
-			first = 0;
-			putchar(',');
+		break;
+
+	case FORMAT_DSV:
+		buf = escape_delimiter(value, delimiter, ESC_CHAR);
+		if (buf == NULL) {
+			fprintf(stderr, "qstat: out of memory\n");
+			exit(1);
+		}
+		buf = convert_feed_chars(buf);
+		if (buf == NULL) {
+			fprintf(stderr, "qstat: out of memory\n");
+			exit(1);
+		}
+		if (resource)
+			printf("%s.%s=%s", name, resource, buf);
+		else
+			printf("%s=%s", name, buf);
+		free(buf);
+		break;
+
+	default:
+		if (one_line) {
+			buf = convert_feed_chars(value);
+			if (buf == NULL) {
+				fprintf(stderr, "qstat: out of memory\n");
+				exit(1);
+			}
+			if (resource)
+				printf("    %s.%s = %s", name, resource, buf);
+			else
+				printf("    %s = %s", name, buf);
+			free(buf);
+		} else {
+			start = strlen(name) + 7; /* 4 spaces + ' = ' is 7 */
+			printf("    %s", name);
```

Comment:
You are right. I tested this scenario with a custom server resource having a lengthy name (>78 chars). resource name supports more than 78 chars but qstat -f doesn't wrap this and print as it is. This can be a possible bug but needs to be discussed in the forum as we have to change the existing behavior. Since we are not dealing with qstat -f output in the present EDD, It is raised as a separate open Jira issue: PP-720.